src/transformers/tokenization_utils.py

```diff
@@ -335,33 +405,49 @@ def additional_special_tokens(self):
             logger.error("Using additional_special_tokens, but it is not set yet.")
         return self._additional_special_tokens
 
+    def _update_if_fast_tokenizer(self, value):
+        if self.is_fast:
+            self._tokenizer.add_special_tokens(value)
```

Comment:
So we can gather all the special token handling in `SpecialTokensMixin`

Reply:
It took me a while to understand that you can access a property from the `PreTrainedTokenizer` that will extend the `SpecialTokensMixin`

Reply:
Agree, `_tokenizer` and `is_fast` doesn't exist on `SpecialTokenMixin`. Moreover, this behaviour is only relevant on FastTokenizer, so using inheritance seems less hacky

Reply:
As this is special case for `PreTrainedTokenizerFast` subclasses, I would rather put this in `PreTrainedTokenizerFast` as it would remove the need to have conditional branching

Reply:
Ok will move this method to the derived class